hbw/app/assets/javascripts/hbw/components/form/file_upload.js.jsx.coffee

```diff
@@ -5,14 +5,19 @@ modulejs.define 'HBWFormFileUpload',
       mixins: [DeleteIfMixin, CallbacksMixin]
 
       getInitialState: ->
-        { value: '', valid: true, files: [], files_count: 0}
+        {value: '', valid: true, files: [], files_count: 0, incoming_files: []}
 
       render: ->
         opts = {
           name: @props.params.name
           onChange: @onChange
         }
 
+        if (@props.value)
+          @state.incoming_files = JSON.parse(@props.value).files
```

Comment:
```props``` should not be modified. So, you set ```incoming_files``` every if you have value in props. Every time. Why?

Reply:
props can be modified by activiti, imprint integrations for example.

Reply:
Maybe I don't understand you, but https://reactjs.org/docs/components-and-props.html#props-are-read-only